src/ruby/ext/grpc/rb_channel.c

```diff
@@ -795,6 +855,12 @@ void Init_grpc_channel() {
   rb_define_method(grpc_rb_cChannel, "initialize_copy",
                    grpc_rb_cannot_init_copy, 1);
 
+#ifdef GPR_SUPPORT_CHANNELS_FROM_FD
+  /* Provides a Ruby constructor for wrapping an existing file descriptor */
+  rb_define_singleton_method(grpc_rb_cChannel, "insecure_create_from_fd",
```

Comment:
instead of not defining the `insecure_create_from_fd ` when building for the platforms that don'e support it, I'd prefer having it's implementation throw `NotImplementedError ` on those platforms.

Can we change this around so that the ruby method always exists, but so that the implementation if it either calls the inner `insecure_create_from_fd` or raises `NotImplementedError`? e.g., move the `ifdef` to the implementation?

Reply:
Done.